src/app/pipes/meql.pipe.spec.ts

```diff
@@ -0,0 +1,356 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+import { MeqlPipe } from './meql.pipe';
+import { TestBed } from '@angular/core/testing';
+import moment from 'moment';
+
+const newline = '\r\n';
+const tab = '\t';
+
+class meqlTestHelper {
+  public static singleConditionTest(
+    pipe: MeqlPipe,
+    field: string,
+    operator: string,
+    value: any
+  ) {
+    /*
+  Expected example:
+
+  (
+    "Field" = "Value"
+  )
+   */
+    const query = {
+      condition: 'and',
+      rules: [{ field, operator, value }],
+    };
+    const line1 = `(${newline}`;
+    const line2 = `${tab}"${query.rules[0].field}" ${query.rules[0].operator} "${query.rules[0].value}"${newline}`;
+    const line3 = ')';
+    return [pipe.transform(query), `${line1}${line2}${line3}`];
+  }
+
+  public static doubleConditionTest(pipe: MeqlPipe, query: any) {
+    /*
+  Expected example:
+
+  (
+    "Field" = "Value"
+    and "Field2" = "Value2"
+  )
+   */
+    const line1 = `(${newline}`;
+    const line2 = `${tab}"${query.rules[0].field}" ${query.rules[0].operator} "${query.rules[0].value}"${newline}`;
+    const line3 = `${tab}${query.condition} "${query.rules[1].field}" ${query.rules[1].operator} "${query.rules[1].value}"${newline}`;
+    const line4 = ')';
+    return [pipe.transform(query), `${line1}${line2}${line3}${line4}`];
+  }
+
+  public static doubleRulesetTest(pipe: MeqlPipe, query: any) {
+    /*
+  Expected example:
+
+  (
+    "Field" = "Value"
+    and "Field2" = "Value2"
+    and (
+        "Field3" = "Value3"
+        or "Field4" = "Value4"
+    )
+  )
+   */
+    const line1 = `(${newline}`;
+    const line2 = `${tab}"${query.rules[0].field}" ${query.rules[0].operator} "${query.rules[0].value}"${newline}`;
+    const line3 = `${tab}${query.condition} "${query.rules[1].field}" ${query.rules[1].operator} "${query.rules[1].value}"${newline}`;
+    const line4 = `${tab}${query.condition} (${newline}`;
+    const line5 = `${tab}${tab}"${query.rules[2].rules[0].field}" ${query.rules[2].rules[0].operator} "${query.rules[2].rules[0].value}"${newline}`;
+    const line6 = `${tab}${tab}${query.rules[2].condition} "${query.rules[2].rules[1].field}" ${query.rules[2].rules[1].operator} "${query.rules[2].rules[1].value}"${newline}`;
+    const line7 = `${tab})${newline}`;
+    const line8 = ')';
+    return [
+      pipe.transform(query),
+      `${line1}${line2}${line3}${line4}${line5}${line6}${line7}${line8}`,
+    ];
+  }
+}
+
+describe('JqlPipe', () => {
```

Comment:
Changing the name of the pipe meant this was not updated, please correct

Reply:
Corrected